### PR TITLE
AZP: Fail if running on a GPU machine without GPU-direct driver

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Detect missing GPU-direct driver

### CODE REVIEW

**Comment by yosefe** [comment]
@avildema can you pls take a look?

File: buildlib/az-helpers.sh [hunk 1]
```diff
@@ -128,6 +128,11 @@ try_load_cuda_env() {
 		az_module_load dev/cuda11.1.1 || have_cuda=no
 		az_module_load dev/gdrcopy2.1_cuda11.1.1 || have_gdrcopy=no
 		num_gpus=$(nvidia-smi -L | wc -l)
+        if [ "$num_gpus" -gt 0 ] && ! [ -f /sys/kernel/mm/memory_peers/nv_mem/version ]
+        then
+            lsmod
```
Comment (avildema) [inline buildlib/az-helpers.sh:133]: for more information may be add `nvidia-smi` or `nvidia-smi -a` and check gpu devices `ls /dev | grep nvidia`

Comment (yosefe) [reply buildlib/az-helpers.sh:133]: this code is run when GPU is found by nvidia-smi, but peer-direct is not found, so why need more nvidia-smi diagnostic?

Comment (avildema) [reply buildlib/az-helpers.sh:133]: suddenly there will be a situation like mine, we have nvidia/cuda docker, we have "/proc/driver/nvidia/version" but we don't have device

Comment (yosefe) [reply buildlib/az-helpers.sh:135]: ok, added

**Comment by yosefe** [comment]
@avildema is it ok now?

**Review by avildema — APPROVED** [review]

**Pull request merged**